pkg/nirmata/resource_cluster_registered.go

```diff
@@ -72,13 +116,137 @@ func resourceClusterRegistered() *schema.Resource {
+				}
+				accessControlListObj = map[string]interface{}{
+					"modelIndex": "AccessControlList",
+					"ownerType":  owner_type,
+					"ownerName":  owner_name,
+					"ownerId":    owner_id,
+				}
+			}
+		}
+	} else {
+		log.Printf("[DEBUG] - owner_info is empty")
+		accessControlListObj = map[string]interface{}{
+			"modelIndex": "AccessControlList",
+			"ownerType":  owner_type,
+		}
+	}
+	return accessControlListObj, nil
+}
+
+func makeAccessControlList(accessControlList []interface{}, users, teams []map[string]interface{}) ([]interface{}, error) {
+	var acObArr = make([]interface{}, 0)
+
+	if accessControlList != nil {
+		for _, ac := range accessControlList {
+			log.Printf("[DEBUG] - ac %v", ac)
+			elem, ok := ac.(map[string]interface{})
+			if ok {
+				entity_type := elem["entity_type"].(string)
+				entity_name := elem["name"].(string)
+				entity_id, err := fetchOwnerId(entity_type, entity_name, users, teams)
+				if err != nil {
+					log.Printf("[ERROR] - No entity of type %s with name %s found in the cluster", entity_type, entity_name)
+					return nil, fmt.Errorf("[ERROR] - Invalid entity '%s/%s' provided", entity_type, entity_name)
+				}
+				acOb := map[string]interface{}{
+					"modelIndex": "AccessControl",
+					"entityType": entity_type,
+					"entityName": entity_name,
+					"permission": elem["permission"],
+					"entityId":   entity_id,
+				}
+				log.Printf("[DEBUG] - Printing accessControlList - acOb %v", acOb)
+				acObArr = append(acObArr, acOb)
+			}
+		}
+	}
+	return acObArr, nil
+}
+
 func resourceClusterRegisteredCreate(d *schema.ResourceData, meta interface{}) error {
 	apiClient := meta.(client.Client)
 	name := d.Get("name").(string)
 	labels := d.Get("labels")
 	endpoint := d.Get("endpoint").(string)
 	clusterType := d.Get("cluster_type").(string)
 
+	users, err := getUsers(apiClient)
+	if err != nil {
+		log.Printf("[ERROR] - failed to retrieve users: %v", err)
+		return fmt.Errorf("[ERROR] - Fetching users failed")
+	}
+	log.Println("[DEBUG] - Printing users")
+	for _, u := range users {
+		for k, v := range u {
+			log.Printf("[DEBUG] - key: %s - value: %s", k, v)
+		}
+	}
+
+	teams, err := getTeams(apiClient)
+	if err != nil {
+		log.Printf("[ERROR] - failed to retrieve teams: %v", err)
+		return fmt.Errorf("[ERROR] - Fetching teams failed")
+	}
+	log.Println("[DEBUG] - Printing teams")
+	for _, t := range teams {
+		for k, v := range t {
+			log.Printf("[DEBUG] - key: %s - value: %s", k, v)
+		}
+	}
+
+	ownerInfo := d.Get("owner_info").([]interface{})
+	log.Printf("[DEBUG] - Printing owner_info from resource %v", ownerInfo)
+
+	accessControlListObj, err := makeAccessControlListObj(ownerInfo, users, teams)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] - Printing accessControlListObj %v", accessControlListObj)
+
+	accessControlList := d.Get("access_control_list").([]interface{})
+	log.Printf("[DEBUG] - Printing accessControlList %v", accessControlList)
```

Comment:
Difference between accessControlList and accessControlListObj is not clear. From names seems that listObj is one element of the List. Guess this is the case?